installer/upgrader/Migrations/Version20220301.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Tools\Migrations;
+
+use OrangeHRM\Tools\Migrations\V5\AttendanceHelper;
+
+class Version20220301
+{
+    protected ?AttendanceHelper $attendanceHelper = null;
+
+    /**
+     * @return AttendanceHelper
+     */
+    public function getAttendanceHelper(): AttendanceHelper
+    {
+        if (is_null($this->attendanceHelper)) {
+            $this->attendanceHelper = new AttendanceHelper();
+        }
+        return $this->attendanceHelper;
+    }
+
+    public function up(): void
+    {
+        //getting all attendance records from data source
+        $attendanceRecordList = $this->getAttendanceHelper()->getAttendanceRecords();
+        foreach ($attendanceRecordList as $attendanceRecords) {
+            $id = $attendanceRecords['id'];
+            $punchOutTimeZone = null;
+            $punchInTimeZone = array_search($attendanceRecords['punch_in_time_offset'], AttendanceHelper::TIMEZONE_MAP);
```

Comment:
```suggestion
        $attendanceRecordList = $this->getAttendanceHelper()->getAttendanceRecords();
        $timezoneMap = array_flip(AttendanceHelper::TIMEZONE_MAP);
        foreach ($attendanceRecordList as $attendanceRecords) {
            $id = $attendanceRecords['id'];
            $punchOutTimeZone = null;
            $punchInTimeZone = $timezoneMap[$attendanceRecords['punch_in_time_offset']];
```

Reply:
i did like this, but i got warning in console. thats y i used to array_search

Reply:
done